pkg/graveler/committed/merge.go

```diff
@@ -321,11 +321,10 @@ func (m *merger) handleConflict(sourceValue *graveler.ValueRecord, destValue *gr
 
 // handleBothKeys handles the case where both source and dest iterators are inside range
 func (m *merger) handleBothKeys(sourceValue *graveler.ValueRecord, destValue *graveler.ValueRecord) error {
-	c := bytes.Compare(sourceValue.Key, destValue.Key)
 	switch {
-	case c < 0: // source before dest
+	case isValue1BeforeValue2(sourceValue, destValue): // source before dest
 		return m.sourceBeforeDest(sourceValue)
-	case c > 0: // dest before source
+	case isValue1BeforeValue2(destValue, sourceValue): // dest before source
```

Comment:
I prefer that the [code will document itself](https://martinfowler.com/bliki/CodeAsDocumentation.html) rather than adding comments for that purpose. In addition, if this functionality is needed in more than, lets say, two places (which you'll soon see needed), you'll need to comment in those places too instead of letting a function do that in your behalf.
Calling `byte.Compare` seems to me like a reasonable price to pay.
I actually forgot to delete the comments for the cases...
Having said that, if there's still great opposition for this change I'll revert it.